enhancements/agent-installer/development-environment.md

```diff
@@ -0,0 +1,305 @@
+breakages in the tests across the different teams.
+
+** The dev-scripts repository clones the [metal-dev-env](https://github.com/metal3-io/metal3-dev-env) 
+repository for most of its Ansible roles. Which means there is a risk of changes in this 
+repository can break things in our repository as well. We have less control over this repository 
+than others inside Openshift.
+
+This is partially mitigated already because the repository is cloned at a specific
+Git SHA. To further this we can add a test in dev-scripts that runs maybe once or twice
+a week that clones the HEAD of this repository. We would be alerted for breakages more
+frequently and we can mitigate them on a case-by-case basis.
+
+
+## Design Details
+
+### Phase 1 (Crawl): Add development scripts into dev-scripts for Agent Installer
+
+1) Add development scripts into dev-scripts for the new agent installer project using 
+existing conventions.
+2) Add base e2e test flow to run the ephemeral agent installer image in Openshift CI.
+3) Write documentation for new developers joining the agent installer team.
+
+In this phase it should also become more clear what scripts and code are overlapping
+between the two teams.
+
+### Phase 2 (Walk): Update shell scripts to Ansible in dev-scripts and use them for Agent Installer
+
+1) Convert the shell scripts [01_install_requirements.sh](https://github.com/openshift-metal3/dev-scripts/blob/master/01_install_requirements.sh) and [02_configure_host.sh](https://github.com/openshift-metal3/dev-scripts/blob/master/02_configure_host.sh) 
+into Ansible roles and playbooks for reuse.
+2) Add in more configuration examples like [config_example.sh](https://github.com/openshift-metal3/dev-scripts/blob/master/config_example.sh) for dev-scripts that define 
+common configurations for OCP installs. 
+3) Use the above examples to define default configurations for VMs (HA, Compact, SNO) 
+and networks (IPV4, IPV6, dualstack) in Ansible.
+
+### Phase 3 (Run): Update existing e2e IPI flows to use the refactored Ansible in dev-scripts
+
+Update the existing e2e IPI flows to use the new Ansible roles and playbooks and debug
+to ensure they work as expected. Doing this in a separate phase will allow us to explore
+new options for the Agent Installer project while maintaining healthy e2e flows for IPI.
+This should allow for more collaboration between the two teams.
+
+
+### Refactored File and Folder Structure
+
+The following is describing general pattern of how to organize these things in the dev-scripts 
+repository. These files and folders will be added alongside existing scripts in the dev-scripts 
+repository. This example includes things that have been already identified as overlaps between 
+the two teams that will need to be added or refactoring. Things may change names or locations
+during the code review process this folder structure is added for clarification.
+
+**File and Folder Structure:**
```

Comment:
I think that here there could be the risk to over-engineering the solution, considering that in future things may evolve differently (for example, integration within the openshift installer). In addition there are parts that seem to overlap with already existing frameworks such as the `ci-operator` and already defined steps at OpenShift level (ie ci-operator steps registry)

Reply:
We'll need a separate step to https://github.com/openshift/release/blob/master/ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh (just running `make` won't cut it), so what's the harm in putting most of the configuration for each topology/network scenario into the dev-scripts repo where developers can also make use of it, and having just 1 configuration option (i.e. which scenario to choose) in the ci-operator steps?

Reply:
This is the route I went with on the rewording of it. Closer to the intention of it all as well.